src/xtopcom/xrpc/xcluster/xcluster_query_manager.cpp

```diff
@@ -16,6 +16,8 @@
 #include "xstore/xtgas_singleton.h"
 #include "xutility/xhash.h"
 #include "xvm/manager/xcontract_address_map.h"
+#include "xmbus/xevent_behind.h"
+#include "xvledger/xvledger.h"
```

Comment:
looks like a separate xrpc/xfull_node module is required to handle full node related issues